clients/http/utils/common.go

```diff
@@ -67,11 +67,10 @@ func makeRequest(req *http.Request) (*http.Response, errors.EdgeX) {
 }
 
 func createRequest(ctx context.Context, httpMethod string, baseUrl string, requestPath string, requestParams url.Values) (*http.Request, errors.EdgeX) {
-	u, err := url.Parse(baseUrl)
+	u, err := url.Parse(baseUrl + requestPath)
```

Comment:
`+` is not equivalent to the `path.Join` previously.  Same change needed in all locations

Also why is this change part of this PR? Has nothing to do with URL escaping?

```suggestion
	u, err := url.Parse(path.Join(u.Path, requestPath))
```

Reply:
We use u.String() to create an HTTP request object
`req, err := http.NewRequest(httpMethod, u.String(), nil)`

But the previous request URL is not expected, it does escape again:
```
func TestUrlStringFunction(t *testing.T) {
	baseUrl := "http://localhost:59980"
	requestPath := "/api/v2/device/name/Modbus-TCP-Device/device-a%2Ftest%3Avalue"

	u1, _ := url.Parse(baseUrl)
	u1.Path = path.Join(u1.Path, requestPath)
	fmt.Println("1.", u1.String())

	u2, _ := url.Parse(baseUrl + requestPath)
	fmt.Println("2.", u2.String())

	u3, _ := url.Parse(path.Join(baseUrl, requestPath))
	fmt.Println("3.", u3.String())
}
```
```
1. http://localhost:59980/api/v2/device/name/Modbus-TCP-Device/device-a%252Ftest%253Avalue
2. http://localhost:59980/api/v2/device/name/Modbus-TCP-Device/device-a%2Ftest%3Avalue
3. http:/localhost:59980/api/v2/device/name/Modbus-TCP-Device/device-a%2Ftest%3Avalue
```

Reply:
Also, `path.Join()` is for file paths and will generate the path differently when running on windows. Not sure it is appropriate here for a URL path.

Reply:
Neverminded ;-)
```
The path package should only be used for paths separated by forward slashes, such as the paths in URLs. This package does not deal with Windows paths with drive letters or backslashes; to manipulate operating system paths, use the path/filepath package.
```

Reply:
https://go.dev/play/p/f3Ig2PcArH3

Reply:
```go
func createRequestWithEncodedData(ctx context.Context, httpMethod string, baseUrl string, requestPath string, data []byte, encoding string) (*http.Request, errors.EdgeX) {
	u, err := parseBaseUrlAndRequestPath(baseUrl, requestPath)
	if err != nil {
		return nil, errors.NewCommonEdgeX(errors.KindServerError, "failed to parse baseUrl and requestPath", err)
	}
```

```go
func parseBaseUrlAndRequestPath(baseUrl, requestPath string) (*url.URL, error) {
	fullPath, err := url.JoinPath(baseUrl, requestPath)
	if err != nil {
		return nil, err
	}
	return url.Parse(fullPath)
}
```

Reply:
@ajcasagrande is there an issue or are you suggesting a refactoring?

Reply:
oh, I saw the issue opened.  please ignore my question.